benchmark/david_benchmark.py

```diff
@@ -0,0 +1,151 @@
+# © Crown Copyright GCHQ
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Benchmark performance of different coreset algorithms on pixel data from an image.
+
+The benchmarking process follows these steps:
+1. Load an input image and downsample it using a specified factor.
```

Comment:
Can you briefly describe _why_ we downsample the image?

Reply:
Have added a brief description:
```
(downsampling reduces the image resolution, 
   i.e., the number of points in the dataset, to reduce the computational load on the
   machine this script is run on. A downsampling factor of 1 corresponds to no 
   downsampling)
```